File: PRACTICA2/deepspace/src/deepspace/SuppliesPackage.java

package deepspace;

class SuppliesPackage {
    private float ammoPower;
    private float fuelUnits;
    private float shieldPower;
    
    SuppliesPackage(float ammoPower,float fuelUnits, float shieldPower){
        this.ammoPower=ammoPower;
        this.fuelUnits=fuelUnits;
        this.shieldPower=shieldPower;
    }
    
    SuppliesPackage(SuppliesPackage s){
        this(s.ammoPower, s.fuelUnits, s.shieldPower);
        
 
    }
    
    public float getAmmoPower(){
        return ammoPower;
    }
    
    public float getFuelUnits(){
        return fuelUnits;
    }
    
    public float getShieldPower(){
        return shieldPower;
    }
    
    @Override
    public String toString(){
        String salida="[SuppliesPackage] -> ammoPower: "+ ammoPower + 
                ", fuelUnits: "+ fuelUnits +
                ", shieldPower: "+ shieldPower;
        return salida;
    }
}

File: PRACTICA4/ruby/SuppliesPackage.rb
#encoding: utf-8
#Esta clase representa a un paquete de suministros para una estación espacial.
module Deepspace
	class SuppliesPackage
		def initialize (ammopower, fuelunits, shieldpower)
			@ammoPower = ammopower
			@fuelUnits = fuelunits
			@shieldPower = shieldpower
		end
		
		def self.newCopy(package)
			new(package.ammoPower, package.fuelUnits, package.shieldPower)
		end
		
		def ammoPower
			@ammoPower
		end
		
		def fuelUnits
			@fuelUnits
		end
		
		def shieldPower
			@shieldPower
		end

		def to_s
			salida="[SuppliesPackage] -> ammoPower: "+ ammoPower.to_s + ", fuelUnits: "+ fuelUnits.to_s + ", shieldPower: "+ shieldPower.to_s
			return salida
		end
	end
end
	

